ocaml/lambda/translcore.ml

```diff
@@ -1888,8 +1894,8 @@ and transl_record ~scopes loc env mode fields repres opt_init_expr =
                blocks containing unboxed float literals.
             *)
             raise Not_constant
-        | Record_inlined (_, _, Variant_extensible)
-        | Record_inlined (Extension _, _, _) ->
+        | Record_inlined (_, _, (Variant_extensible | Variant_with_null))
```

Comment:
The new case should be impossible.

Reply:
That's right, so it's handled the same as other impossible cases. Same for other comments.